resources/files/install-tarballs/extract_all.rb

```diff
@@ -0,0 +1,70 @@
+# Idea from https://stackoverflow.com/questions/856891/unzip-zip-tar-tag-gz-files-with-ruby
+# Altered using https://github.com/puppetlabs/puppet/blob/983154f7e29a2a50d416d889a6fed012b9b12399/lib/puppet/module_tool/tar/mini.rb
+#
+# WARNING - This extraction blindly assumes the tarballs are sane and correct. Unlike the `.../module_tool/tar/mini.rb` implementation, this
+# script does not do any checks:
+# - Symlinks within the tar will throw an error
+# - We assume that all files in the tar have relative paths. If there is an absolute path in the tarball it will 'escape' the intended extract directory
+# - Avoids extra disk IO (e.g. expand_path) as the tarballs will have 30,000+ files inside
+
+require 'fileutils'
+require 'rubygems/package'
+require 'zlib'
+
+TAR_LONGLINK = '././@LongLink'
+
+def logmessage(message)
+  puts message if ENV['BOLT_DEBUG']
+end
+
+# From lib/puppet/module_tool/tar/mini.rb
+EXECUTABLE = 0755
+NOT_EXECUTABLE = 0644
+USER_EXECUTE = 0100
+
+def sanitized_mode(old_mode)
+  # From lib/puppet/module_tool/tar/mini.rb
+  old_mode & USER_EXECUTE != 0 ? EXECUTABLE : NOT_EXECUTABLE
+end
+
+def unpack(sourcefile, destdir, _)
```

Comment:
nit: I find this code to be hard to read without using parenthesis for method arguments. 

I think updating everything like:
```ruby
dest = File.join destdir, entry.read.strip
````
to 
```ruby
dest = File.join(destdir, entry.read.strip)
```
would improve readability.

Reply:
As the commit says, this was a copy paste :)

Modified to suit